File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        dp = [0] * (len(nums) - k + 1)
        dp[0] = sum(nums[:k])
        for i in range(1, len(nums)-k+1): 
            dp[i] = dp[i-1] + nums[i+k-1] - nums[i-1]
        return max(dp) / float(k)

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double ksum = 0;
        for (int i = 0; i < k; i++) {
            ksum += nums[i];
        }
        double ans = ksum / k;
        for (int i = k; i < nums.size(); i++) {
            ksum += nums[i] - nums[i-k];
            ans = max(ans, double(ksum / k));
        }       
        return ans;
    }
};